### PR TITLE
fix(sidebar): fix toggle mute dispatch

### PR DESCRIPTION
**Pull request by jasonwoodland**
**What this PR does** 📖
- update `$store.dispatch` calls for toggling audio/video muting in sidebar so that the buttons behave correctly.
- fix audio stream indicator visibility logic in media/user component so it only shows the audio stream indictor when unmuted.

**Which issue(s) this PR fixes** 🔨
AP-1581, AP-1578, AP-1466

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by Jekrimo — APPROVED** [review]

**Comment by phillsatellite** [comment]
Tested: Found a crash when trying to make a phone call Browsers used were Chrome and Brave 🚀 
https://user-images.githubusercontent.com/93608357/169329424-97b31c69-bfe5-42f8-9802-f4fcf69ecb17.mov

**Comment by jasonwoodland** [comment]
@phillsatellite thanks for spotting it! Fixed in last commit

**Comment by stavares843** [comment]
thanks @jasonwoodland 🔨

**Comment by phillsatellite** [comment]
Retested: Crash is gone! Functionality was good on my end the only differences I noticed from this branch and dev is the sound feedback that happens when the user hits mute or video

https://user-images.githubusercontent.com/93608357/169571403-b4897233-06cb-49aa-a95a-87d6fcad55c2.mov

File: components/views/media/user/User.html [hunk 1]
```diff
@@ -28,7 +28,7 @@
       type="random"
       :seed="user.address"
       :size="$device.isMobile ? 36 : 65"
-      :class="`${!audioStream ? 'calling' : ''}`"
+      :class="`${audioStream ? 'calling' : ''}`"
```
Comment (josephmcg) [inline components/views/media/user/User.html:31]: this has some unnecessary wrappers
`:class='{calling : audioStream}'`

**Review by josephmcg — CHANGES_REQUESTED** [review]
mic and video toggles no longer work
image

File: components/views/media/user/User.html (hunk 1 above)
Comment (jasonwoodland) [reply components/views/media/user/User.html]: fixed the class

Comment (jasonwoodland) [reply components/views/media/user/User.html]: I can't reproduce the bug you're getting with the toggle buttons

**Comment by stavares843** [comment]
rebase wasn't working, added latest dev here to update Cypress

**Review by phillsatellite — APPROVED** [review]

**Review by Clayton215 — APPROVED** [review]

**Review by josephmcg — COMMENTED** [review]
The mute/video toggles don't work while you are not in a call. If this is intended functionality, then this is ok

**Comment by jasonwoodland** [comment]
@stavares843 I made some other changes to sync the muted state in the sidebar with the active calls. (see [234b7b7](https://github.com/Satellite-im/Core-PWA/pull/3236/commits/234b7b7dc533f0d7516544f0e275953431b66f79)) So please re-review!

File: store/audio/actions.ts [hunk 2]
```diff
@@ -7,25 +7,24 @@ export default {
...
+      root: true,
+    })
+
+    if (call) {
+      if (muted) {
+        call.mute({ kind: 'audio' })
+      } else {
+        call.unmute({ kind: 'audio' })
+      }
     }
```
Comment (josephmcg) [inline store/audio/actions.ts:27]: we typically use guard clauses, could you rewrite?
`if (!call) {
        return
      }
      if (muted) {
        call.mute({ kind: 'audio' })
        return
      }
      call.unmute({ kind: 'audio' })
`

File: components/views/media/actions/Actions.vue [hunk 3]
```diff
@@ -54,11 +54,15 @@ export default Vue.extend({
      */
     toggleMute(kind = 'audio') {
```
Comment (josephmcg) [inline components/views/media/actions/Actions.vue:55]: this automatic value is never used. could we use keyof PeerMutedState for kind param

File: components/views/navigation/sidebar/controls/Controls.vue [hunk 4]
```diff
@@ -51,17 +49,9 @@ export default Vue.extend({
      * @example
      */
     async toggleMute(kind = 'audio') {
```
Comment (josephmcg) [inline components/views/navigation/sidebar/controls/Controls.vue:51]: could do same improvement here for types. also change self closing template at the top

**Review by josephmcg — CHANGES_REQUESTED** [review]
change the first line of Actions.vue to this. I've noticed linting breaks if the template is self closing
image

File: store/audio/actions.ts (hunk 2 above)
Comment (josephmcg) [reply store/audio/actions.ts]: double check your implementation of this. we want to eliminate else. we also want to return if `!call`, right?

**Review by josephmcg — CHANGES_REQUESTED** [review]
check previous comment

**Comment by jasonwoodland** [comment]
@josephmcg fixed!

File: store/audio/actions.ts (hunk 2 above)
Comment (jasonwoodland) [reply store/audio/actions.ts]: my bad, fixed

File: store/audio/actions.ts [hunk 5]
```diff
@@ -7,25 +7,25 @@ export default {
...
+    if (!call) {
       return
     }
-    if (call) call.mute({ kind: 'audio' })
-    dispatch('sounds/playSound', Sounds.MUTE, { root: true })
+
+    if (muted) {
+      call.mute({ kind: 'audio' })
+    }
+    call.unmute({ kind: 'audio' })
```
Comment (josephmcg) [inline store/audio/actions.ts:28]: guard clause still funky, missing return

**Review by josephmcg — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
the video toggle doesn't work if you are not on a call, works on dev

Captura de ecrã 2022-05-26, às 18 48 13


btw I noticed when hovering it says ''Toggle'' but that issue is on dev as well so I added a ticket [here](https://satellite-im.atlassian.net/browse/AP-1637)

**Comment by WanderingHogan** [comment]
@stavares843 I don't think we actually want to allow the video toggle if you are not on a call since we are forcing all calls to start as audio. I think another ticket to disable the button when not on a call might be warranted, what do you think?

**Comment by stavares843** [comment]
sounds good, thanks @WanderingHogan 🔨

**Comment by stavares843** [comment]
added ticket [here](https://satellite-im.atlassian.net/browse/AP-1647)

**Review by stavares843 — APPROVED** [review]

**Pull request merged**